src/components/tabs/CdrTabs.vue

```diff
@@ -89,276 +23,244 @@ export default defineComponent({
-      }
-      activeTabIndex.value = newIndex;
-      updateUnderline();
-      cdrTabsHeaderEl.value.children[activeTabIndex.value].focus();
     }
-    const rightArrowNav = debounce(function handleRightArrow() {
-      const nextTab = getNextTab(activeTabIndex.value + 1);
-      if (nextTab !== -1) {
-        changeTab(nextTab);
-      }
-    }, 300, { leading: true, trailing: false })
-
-    const leftArrowNav = debounce(function handleLeftArrow() {
-      const previousTab = getPreviousTab(activeTabIndex.value - 1);
-      if (previousTab !== -1) {
-        changeTab(previousTab);
-      }
-    }, 300, { leading: true, trailing: false })
+})
+
+const slots = useSlots();
+const slottedTabs = slots.default()[0].children?.length ? slots.default()[0].children : slots.default();
+const baseClass = 'cdr-tabs';
+
+//Refs
+const tabs = ref(slottedTabs.map((tab) => ({ name: tab.props.name, disabled: tab.props.disabled })));
+const selectedTabName = ref(null);
+const selectedIndex = ref(null);
+const headerOverflow = ref(false);
+const headerWidth = ref(0);
+const tabElements = ref([]);
+const tablist = ref(null);
+const containerEl = ref(null);
+const overflowLeft = ref(false);
+const overflowRight = ref(false);
+const underlineOffsetX = ref(0);
+const underlineWidth = ref(0);
+
+provide("selectedTabName", selectedTabName);
+
+//Computed
+const modifierClass = computed(() => modifyClassName('cdr-tabs', props.modifier));
+const sizeClass = computed(() => props.size && modifyClassName('cdr-tabs', props.size));
+const underlineStyle = computed(() => ({
+    transform: `translateX(${underlineOffsetX.value}px)`,
+    width: `${underlineWidth.value}px`,
+}));
+const gradientLeftStyle = computed(() => {
+    const gradient = `linear-gradient(to left, rgba(255, 255, 255, 0), ${props.backgroundColor})`;
+    return {
+    background: gradient,
+    };
+});
+const gradientRightStyle = computed(() => {
+    const gradient = `linear-gradient(to right, rgba(255, 255, 255, 0), ${props.backgroundColor})`;
+    return {
+    background: gradient,
+    };
+});
 
-    const calculateOverflow = () => {
-      let containerWidth = 0;
-      if (containerEl.value) {
+ const calculateOverflow = () => {
+    let containerWidth = 0;
+    if (containerEl.value) {
         containerWidth = containerEl.value.offsetWidth;
-      }
-      headerOverflow.value = headerWidth.value > containerWidth;
-      if (headerOverflow.value) {
+    }
+    headerOverflow.value = headerWidth.value > containerWidth;
+    if (headerOverflow.value) {
         // Get Scroll Position
-        const scrollX = cdrTabsHeaderEl.value.parentElement.scrollLeft;
+        const scrollX = tablist.value.scrollLeft;
         overflowLeft.value = scrollX > 1;
         overflowRight.value = (scrollX + 1) < (headerWidth.value - containerWidth);
-      } else {
+    } else {
         overflowLeft.value = false;
         overflowRight.value = false;
-      }
     }
-    const updateUnderline = () => {
```

Comment:
updateUnderline was migrated from the old version of tabs. I just renamed some variables to account for the change in markup.